resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/resources/ResourcesPlugin.java

```diff
@@ -475,7 +483,12 @@ private static String getSystemEncoding() {
 	 * @return the single instance of this plug-in runtime class
 	 */
 	public static ResourcesPlugin getPlugin() {
-		return plugin;
+		ResourcesPlugin p = plugin;
```

Comment:
This is wrong as it will prevent the workspace from become available unless someone calls this static method

Reply:
i added `new Thread(ResourcesPlugin::getPlugin, "Async ResourcesPlugin start").start();` to start() - ok?

Reply:
Not really why do you do the init in this method and not in the activate that makes no sense to me at all...if you want to defer the tracking to happen in a different thread (for whatever reason) why not simply wrap the open call in a new thread right there?

Reply:
Other plugins cal the static as soon as the Resourceplugin is started and a synchronize needs to wait for completion to finish.

Reply:
This is most likely not correct, it is the other way round. A plugin is starting, calling the static method and this triggers the activator to run and this then is automatically synchronized by the classloader lock already, so why one wants to break this behavior and instead add another synchronization.
These static access is actually bad and has lead to numerous problem in the past, so now requiring this for the startup of the plugin is wrong in many ways and you probably want to better sync on the project loading if projects are accessed.

Reply:
Then this PR can't go in in this form. Relying on a static method call to init code that the Activator must perform to "fix" other plugins is simply not acceptable.

Reply:
You still don't understand this PRs changes: It only moves the existing initialization code into a static method which is nothing bad. If you know a better pattern please show that it works. There are enough tests that fail when you get it wrong.

Reply:
The previous code has worked correctly and as you see test fail if you do it work, your are just hiding the problem here because test call that static method, so something is flawed here but moving the init into a completely unrelated static method is defiantly not the solution.

So this all circles back to the useage of the forkjoin pool as you obviously don't want to fork and **join(!)** what you so with .get() ... maybe you want to delay the computation and result retrieval to a later stage and the `CompletableFuture` might be a good choice who knows.

Reply:
Not that it changes anything but i made the initialized() a non static, just to please you.

Reply:
This does not address the issues I mentioned here... If you want to do things async then perform them async.

Reply:
I don't understand any of your recent comments. The initialization does happens async and it fullfills all contracts. and of cause the result of the workspace read has to be joined before using it. Please stop wasting my time with arbitrary requests. Show that the implementation does fail somehow if you are convinced it can.

Reply:
> BTW the trick is that the ResourcesPlugin does not have a single real non-static public method. All methods call the static getPlugin() first.

And this is where your misunderstanding starts from, *any classload* will trigger the plugin and it has public exported classes (so any import/use of such class will trigger it) and it has extension points so any usage of extension points will trigger it, so your assumption that the only entrypoint are static methods is wrong.

Reply:
show me how that makes an issue.

Reply:
> show me how that makes an issue.

The Activator is the "synchronized initializer" of the Resources Bundle, synchronizing on the tracker open is prone to deadlocks, we have had this problem when migrating to the tracker approach and similar attempts to your where evaluated and caused problems. The dangerous part is that it *might* work when there is no workspace chooser, and suddenly fails when there is one, it might even work for a particular configuration (e.g. tests, a local workspace, ...) and deadlock once you install another plugin or use another combination. And as this is the most crucial part of Eclipse IDE I don't want to do any experiments just to save a few milliseconds if even not the problem is well described or understood.

So maybe it would better if you try to describe the issue you see or the problem you try to solve instead on insist on a hack.

Reply:
show me that THIS implementation makes an error, not yours. It does not even be a junit. just put breakpoints such that it fails if executed in a predefined order of your choice. But it boils down to that synchronize() just works: only one thread will do the init without waiting and all others wait.
The problem is simple that Eclipse on Windows needs >10 sec to start and >2 of them are caused by loading the projects.
![image](https://github.com/eclipse-platform/eclipse.platform/assets/51790620/d5ceec5c-f949-4c76-89cd-04090cac63c8)
loading them in parallel speeds that  up.

Reply:
> show me that THIS implementation makes an error, not yours. 

If you show me that MY implementation makes an error and not YOURS I might be convinced... longer loading times are not an error, you can for sure make the loading IO of `MarkerManager` parallel, but not by changing the Activator.

Reply:
Optimizing loading time of IDE is a valid and desired topic. Giving only spurious arguments does not help and contradict our code of conduct.

Reply:
Claiming other wasting your time and their implementation is failed is also "contradict our code of conduct" .. I gave you several insight and ways to archive your goal differently so please don't claim any contradiction anytime someone does not agree with you.

Reply:
@jukzi : startup performance and functional integrity of code are two different things.

Independently how fast this PR will make startup for some use case, **if** it will be broken for other use case, it is not acceptable.

And as you propose the change, it is your duty to proof it is safe and works, assuming current state is not broken (which is the case).